lib/datadog/tracing/contrib/utils/quantization/http.rb

```diff
@@ -19,7 +19,9 @@ module HTTP
             def url(url, options = {})
               url!(url, options)
             rescue StandardError
-              options[:placeholder] || PLACEHOLDER
+              placeholder = options[:placeholder] || PLACEHOLDER
+
+              options[:base] == :exclude ? placeholder : "#{base_url(url)}/#{placeholder}"
             end
 
             def base_url(url, options = {})
```

Comment:
Maybe another way of doing it would be to escape the url before trying to parse it with something like this:
```suggestion
            def base_url(url, options = {})
              parser = URI::RFC2396_Parser.new
              URI.parse(parser.escape_url(url)).tap to |uri|
```
With this url `http://example.com/|`, you get this parsable escaped_url `http://example.com/%7C`, and you don't have use the regexp in the rescue block.

Reply:
This would URL escape the whole URL, only for the non-base part to be thrown away, which seems a bit wasteful especially for longer `url`.

That said, your note makes me think about just using the regex in all cases instead of going through "full parsing" (which is a regex too), as it would save a couple of allocs and string walking.

Also, while unlikely, the regex should probably look more like [the beginning of this one](https://github.com/ruby/ruby/blob/4b1504ae0a5118153f8b47d7bd0fff7e61553d17/lib/uri/rfc3986_parser.rb#L5).

Reply:
That's very true! Also yes the regexp from the stdlib is probably more precise. Thank's for the answer

Reply:
Thanks for your feedback! I made the changes as described.